test/run_tests.pl

```diff
@@ -62,7 +62,9 @@ BEGIN
     @tests = map { abs2rel($_, rel2abs(curdir())); } @tests;
 
     my $harness = $TAP_Harness->new(\%tapargs);
-    $harness->runtests(sort @tests);
+    my $ret = $harness->runtests(sort @tests);
+
+    exit $ret->has_errors if (ref($ret) eq "TAP::Parser::Aggregator");
```

Comment:
Note that Test::Harness::runtests doesn't return anything, i.e. in absence of TAP::Harness $ret is `undef`. As commit message says in case when non-zero exit is needed [in absence of TAP::Harness] Test::Harness::runtests will do it for you. So that exit $ret will always be exit(0) in absence of TAP::Harness, but that's what perl would do anyway...